Sources/Network/NetworkStack.swift

```diff
@@ -9,5 +9,8 @@
 import Foundation
 
 public protocol NetworkStack {
-    func fetch<R: NetworkResource>(resource: R, completion: @escaping Network.CompletionClosure) -> Cancelable
+    associatedtype Remote
+
+    func fetch<R: NetworkResource>(resource: R, _ completion: @escaping Network.CompletionClosure<R.Remote>)
```

Comment:
I think we can remove these anonymous parameters, because we can use trailing closures even if they are named, and when using variables it isn't so weird 😉